### PR TITLE
feat(fragment): add new fragment type `Now` and send barriers to now …

### PR DESCRIPTION
**Pull request by soundOfDestiny**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- Summarize your change (**mandatory**)
Add a new fragment type `Now` (i.e., has now) in order that we can send barriers to actors who have `NowExecutor` .
- Describe clearly one logical change and avoid lazy messages (optional)
After this PR, we will send barriers to actors who have either `NowExecutor` or `SourceExecutor`.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
#6889

### CODE REVIEW

File: src/meta/src/manager/catalog/fragment.rs [hunk 1]
```diff
@@ -79,8 +79,8 @@ pub struct ActorInfos {
     /// node_id => actor_ids
     pub actor_maps: HashMap<WorkerId, Vec<ActorId>>,
 
-    /// all reachable source actors
-    pub source_actor_maps: HashMap<WorkerId, Vec<ActorId>>,
+    /// all reachable source and now actors
+    pub source_now_actor_maps: HashMap<WorkerId, Vec<ActorId>>,
```
Comment (BugenZhao) [inline src/meta/src/manager/catalog/fragment.rs:83]: How about using a more general name here? Like `barrier_inject_actor_maps` or `inject_barrier..`.

**Review by BugenZhao — COMMENTED** [review]
Rest LGTM!

File: src/meta/src/manager/catalog/fragment.rs (hunk 1 above)
Comment (soundOfDestiny) [reply src/meta/src/manager/catalog/fragment.rs:83]: > How about using a more general name here? Like `barrier_inject_actor_maps` or `inject_barrier..`.

fixed

**Review by BugenZhao — APPROVED** [review]

File: proto/stream_plan.proto [hunk 2]
```diff
@@ -557,6 +557,7 @@ enum FragmentTypeFlag {
   SOURCE = 1;
   MVIEW = 2;
   SINK = 4;
+  NOW = 8;
```
[the remark this answers is not in the document]
Comment (soundOfDestiny) [reply proto/stream_plan.proto:560]: > 😿

https://github.com/risingwavelabs/risingwave/pull/6524#discussion_r1030074898
https://github.com/risingwavelabs/risingwave/pull/6524#discussion_r1030076388

**Pull request merged**